x/gamm/keeper/multihop_test.go

```diff
@@ -55,32 +65,55 @@ func (suite *KeeperTestSuite) TestBalancerPoolSimpleMultihopSwapExactAmountIn()
 						TokenOutDenom: "baz",
 					},
 				},
+				poolFee:           []sdk.Dec{poolDefaultSwapFee, poolDefaultSwapFee},
 				tokenIn:           sdk.NewCoin("foo", sdk.NewInt(100000)),
 				tokenOutMinAmount: sdk.NewInt(1),
 			},
+			coinA:             sdk.NewCoin("foo", sdk.NewInt(1000000000)),
+			coinB:             sdk.NewCoin("uosmo", sdk.NewInt(1000000000)),
+			coinC:             sdk.NewCoin("baz", sdk.NewInt(1000000000)),
+			reducedFeeApplied: true,
+			expectPass:        true,
+		},
+		{
+			name: "Swap - [foo -> uosmo](pool 1) - [uosmo -> baz](pool 2) with a half fee applied, (pool 1) 1 percent fee, (pool 2) 10 percent fee",
```

Comment:
q: do we have a test where one route has a reduced fee but the next one does not or vice versa?

Reply:
So I might be misunderstanding your question, but the reduced fee is an all or nothing deal. If the route is greater than two hops, the reduced fee does not get applied. Please lmk if this addresses your concern

Reply:
Makes sense but do we have a test to cover this case?

That is, more than 2 hoops AND the first hop is osmo routed BUT the half fee is not applied

Reply:
Agreed, will add this!